File: emailsender/languages/dart/relay_policy.dart
class RelayPolicy {
  final bool authorized, openRelay, spoofFrom, privacyMode;
  const RelayPolicy({this.authorized = true, this.openRelay = false, this.spoofFrom = false, this.privacyMode = false});
  void validate() {
    if (openRelay || spoofFrom) throw StateError('Open relays and From spoofing are disabled');
    if (!authorized) throw StateError('Relay authorization is required');
  }
}

File: emailsender/languages/php/RelayPolicy.php
<?php
namespace BizX\EmailSender;

final class RelayPolicy {
    public function __construct(public bool $authorized=true, public bool $openRelay=false, public bool $spoofFrom=false, public bool $privacyMode=false) {}
    public function validate(): void {
        if ($this->openRelay || $this->spoofFrom) throw new \RuntimeException('Open relays and From spoofing are disabled');
        if (!$this->authorized) throw new \RuntimeException('Relay authorization is required');
    }
}
